cmd/daprd/main.go

```diff
@@ -150,287 +144,107 @@ func main() {
 
 	err = rt.Run(
 		runtime.WithSecretStores(
-			secretstores_loader.New("kubernetes", func() secretstores.SecretStore {
-				return sercetstores_kubernetes.NewKubernetesSecretStore(logContrib)
-			}),
-			secretstores_loader.New("azure.keyvault", func() secretstores.SecretStore {
-				return keyvault.NewAzureKeyvaultSecretStore(logContrib)
-			}),
-			secretstores_loader.New("hashicorp.vault", func() secretstores.SecretStore {
-				return vault.NewHashiCorpVaultSecretStore(logContrib)
-			}),
-			secretstores_loader.New("aws.secretmanager", func() secretstores.SecretStore {
-				return secretmanager.NewSecretManager(logContrib)
-			}),
-			secretstores_loader.New("aws.parameterstore", func() secretstores.SecretStore {
-				return parameterstore.NewParameterStore(logContrib)
-			}),
-			secretstores_loader.New("gcp.secretmanager", func() secretstores.SecretStore {
-				return gcp_secretmanager.NewSecreteManager(logContrib)
-			}),
-			secretstores_loader.New("local.file", func() secretstores.SecretStore {
-				return secretstore_file.NewLocalSecretStore(logContrib)
-			}),
-			secretstores_loader.New("local.env", func() secretstores.SecretStore {
-				return secretstore_env.NewEnvSecretStore(logContrib)
-			}),
+			secretstores_loader.New("kubernetes", sercetstores_kubernetes.NewKubernetesSecretStore(logContrib)),
```

Comment:
Wait. Would this create an instance of every component, even unused ones?

/cc @yaron2 ?

Reply:
@artursouza That is correct. The purpose of the funcs to create instances (factory functions) is so that a component is create for each configuration that uses that component type. This change would mean that a single instance would be returned for all component configurations and it would be a pretty bad race condition.